services/src/main/scala/cromwell/services/healthmonitor/impl/CommonMonitoredSubsystems.scala

```diff
@@ -0,0 +1,39 @@
+package cromwell.services.healthmonitor.impl
+
+import cromwell.core.docker.{DockerCliClient, DockerCliKey}
+import cromwell.services.EngineServicesStore
+import cromwell.services.healthmonitor.HealthMonitorServiceActor.{MonitoredSubsystem, OkStatus, SubsystemStatus}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait CommonMonitoredSubsystems {
+  implicit val ec: ExecutionContext
+
+  lazy val DockerHub = MonitoredSubsystem("DockerHub", checkDockerhub _)
+  lazy val EngineDb = MonitoredSubsystem("Engine Database", checkEngineDb _)
+
+  /**
+    * Demonstrates connectivity to DockerHub by periodically pulling a small image. Remove the image afterwards in
+    * order to ensure that subsequent attempts aren't showing a false positive
+    */
+  private def checkDockerhub(): Future[SubsystemStatus] = {
+    Future {
+      // Using a tiny docker image the user is very unlikely to be actively using
+      val alpine = DockerCliKey("alpine", "latest") // FIXME: make a clone of alpine in our dockerhub and use that
+
+      val res = for {
+        p <- DockerCliClient.pull(alpine)
+        r <- DockerCliClient.rmi(alpine)
+      } yield r
+
+      res.get
+    } map { _ => OkStatus }
+  }
+
+  /**
+    * Demonstrates connectivity to the engine database by periodically making a small query
+    */
+  private def checkEngineDb(): Future[SubsystemStatus] = {
+    EngineServicesStore.engineDatabaseInterface.queryDockerHashStoreEntries("DOESNOTEXIST") map { _ => OkStatus }
+  }
```

Comment:
I intentionally didn't bother checking the metadata db. Reasoning was that realistically anyone who cares about this is currently still sharing the DB. Also, since this whole feature is mainly for workbench, once we make a new metadata impl for caas, that check should go over in the workbench impl. Or something like that. Too many question marks (One could say `???`) at the moment to make me want to bother.